### PR TITLE
close response body if fail to send downstream

### PR DESCRIPTION
**Pull request by joesis**
For https://github.com/getlantern/lantern-internal/issues/909#issuecomment-306480036

Test code is pending as it requires replicating most of the `doTest` code.

### CODE REVIEW

File: proxy_http.go [hunk 1]
```diff
@@ -174,7 +174,13 @@ func (ic *httpInterceptor) writeResponse(downstream io.Writer, resp *http.Respon
 	} else {
 		resp = ic.onResponse(prepareResponse(resp, belowHTTP11))
 	}
-	return resp.Write(out)
+	err := resp.Write(out)
```
Comment (oxtoacart) [inline proxy_http.go]: Nice find!

**Comment by joesis** [comment]
Still couldn't come up with a test to verify the effect of this PR without exposing internal state (the response itself). The difference of `MemStats` before and after handling hundreds of requests are not significantly changed by this PR.

I'll add the test anyway to cover a few other cases.

**Comment by joesis** [comment]
Ok added the test. It's unclear whether this PR helps https://github.com/getlantern/lantern-internal/issues/909#issuecomment-306480036, but it does something.

@oxtoacart could you have a look?

File: proxy_http.go [hunk 2]
```diff
@@ -118,14 +118,15 @@ func (ic *httpInterceptor) processRequests(remoteAddr string, req *http.Request,
 
 	first := true
 	for {
+		ic.onRequest(req)
```
Comment (oxtoacart) [inline proxy_http.go]: You need to capture the return value of onRequest.

Comment (oxtoacart) [inline proxy_http.go]: I'll just go ahead and fix this.

**Comment by oxtoacart** [comment]
@joesis I fixed the `ic.OnRequest` thing and it LGTM otherwise.

**Comment by joesis** [comment]
Nice catch @oxtoacart . Thanks for reviewing!

**Pull request merged**